modules/terraform/format.go

```diff
@@ -14,15 +14,32 @@ func FormatArgs(options *Options, args ...string) []string {
 	terraformArgs = append(terraformArgs, FormatTerraformVarsAsArgs(options.Vars)...)
 	terraformArgs = append(terraformArgs, FormatTerraformArgs("-var-file", options.VarFiles)...)
 	terraformArgs = append(terraformArgs, FormatTerraformArgs("-target", options.Targets)...)
+	terraformArgs = append(terraformArgs, FormatTerraformLockAsArgs(options.Lock, options.LockTimeout)...)
 	return terraformArgs
 }
 
-// FormatTerraformVarsAsArgs formats the given variables as command-line args for Terraform (e.g. of the format
-// -var key=value).
+// FormatTerraformVarsAsArgs formats the given variables as command-line args for Terraform
+// (e.g. into the format -var key=value).
 func FormatTerraformVarsAsArgs(vars map[string]interface{}) []string {
 	return formatTerraformArgs(vars, "-var")
 }
 
+// FormatTerraformLockAsArgs formats the lock and lock-timeout variables
+// -lock, -lock-timeout
+func FormatTerraformLockAsArgs(lockCheck bool, lockTimeout string) []string {
```

Comment:
I think there's a bug here: this method will be called for all terraform commands, but not all of them support the `-lock` parameter. I think you'll need this method to know what Terraform command is being called and only add it self to those that support locking. See https://github.com/gruntwork-io/terragrunt/blob/master/config/config_helpers.go#L27.